src/components/Select2.vue

```diff
@@ -72,64 +73,75 @@ export default {
     };
   },
   methods: {
-    toggleSelect: function(init) {
+    toggleSelect(init) {
       if (!init) {
         this.$refs.select.classList.toggle('active');
 
         Fliplet.Widget.autosize();
       }
     },
-    setOption: function(value, init) {
+    setOption(value, init) {
       this.setedOption = this.getSelectedOptionLabel(value);
-      this.$emit('onSelect', value);
+      this.$emit('update:selectedOption', value);
       this.toggleSelect(init);
     },
-    getSelectedOptionValue: function(value) {
+    getSelectedOptionValue(value) {
       if (!this.optionValueKey) {
         return value;
       }
 
       return value && value[this.optionValueKey];
     },
-    getSelectedOptionLabel: function(value) {
+    getSelectedOptionLabel(value) {
       if (!this.optionLabelKey) {
         return value;
       }
 
       return value && value[this.optionLabelKey];
     },
-    search: function() {
+    search(init) {
+      const optionsCopy = _.cloneDeep(this.options);
+
       if (this.customSearch) {
-        this.selectOptions = this.options.filter(option => {
-          return this.customSearch(this.searchValue, option);
-        });
+        if (this.selectWithGroups && !init) {
+          this.selectOptions.forEach((group, index, originArray) => {
+            originArray[index].options = optionsCopy[index].options.filter(option => {
+              return this.customSearch(this.searchValue, option);
+            });
+          });
+        } else {
+          this.selectOptions = optionsCopy.filter(option => {
+            return this.customSearch(this.searchValue, option);
+          });
+        }
 
         return;
       }
 
       this.defaultSearch(this.searchValue);
     },
-    optionView: function(data) {
+    optionView(data) {
       if (this.customOptionView) {
         return this.customOptionView(data);
       }
 
       return data;
     },
-    defaultSearch: function(value) {
+    defaultSearch(value) {
+      const optionsCopy = _.cloneDeep(this.options);
       this.selectOptions = [];
 
       if (!value) {
-        this.selectOptions = [...this.options];
+        this.selectOptions = [...optionsCopy];
```

Comment:
We only need to clone `this.options` in this `if` condition.

```suggestion
        this.selectOptions = _.cloneDeep(this.options);
```